routing/http/contentrouter/contentrouter.go

```diff
@@ -118,7 +111,24 @@ func (c *contentRouter) FindProvidersAsync(ctx context.Context, key cid.Cid, num
 
 	ch := make(chan peer.AddrInfo, len(results))
 	for _, r := range results {
-		ch <- r
+		if r.GetProtocol() == types.BitswapProviderID {
+			result, ok := r.(*types.ReadBitswapProviderRecord)
+			if !ok {
+				logger.Warnw("problem casting find providers result of type", types.BitswapProviderID)
```

Comment:
Can we make this an error? This would indicate a critical bug IMO

Reply:
It must be a panic, right? This methods doesn't return errors

Reply:
i mean log at error level